python/python/lance/fragment.py

```diff
@@ -34,6 +38,48 @@ def __reduce__(self):
         ds = LanceDataset(self._ds.uri, self._ds.version)
         return LanceFragment, (ds, self.fragment_id)
 
+    @staticmethod
+    def create(
+        dataset_uri: Union[str, Path],
+        fragment_id: int,
+        data: pa.Table,
+        schema: Optional[pa.Schema] = None,
+        max_rows_per_group: int = 1024,
+    ) -> LanceFragment:
+        """Create a new fragment from the given data.
+
+        This can be used if the dataset is not yet created.
+
+        Parameters
+        ----------
+        dataset_uri: str
+            The URI of the dataset.
+        fragment_id: int
+            The ID of the fragment.
+        data: pa.Table
+            The data to write to this fragment.
+        schema: pa.Schema, optional
+            The schema of the data. If not specified, the schema will be inferred from the data.
+        """
+        if isinstance(data, pd.DataFrame):
+            reader = pa.Table.from_pandas(data, schema=schema).to_reader()
+        elif isinstance(data, pa.Table):
+            reader = data.to_reader()
+        elif isinstance(data, pa.dataset.Dataset):
+            reader = pa.dataset.Scanner.from_dataset(data).to_reader()
```

Comment:
if the user passes in a LanceDataset it will fall into this case and then fail.
Instead of using the static method, use the Dataset.scanner() (or maybe `to_scanner()`) API (see the other lance methods for this).

The reason is that `pa.dataset.Scanner.from_dataset(...)` ends up referring to Dataset private internals specific to the C++ pyarrow implementation (e.g., CDataset or smth).

Reply:
Oh, i was re-using the same code in `write_dataset`.  Can make it a `to_scanner()` i guess.